File: packages/firestore/ios/Plugin/Classes/Results/AddDocumentResult.swift
import Foundation
import FirebaseFirestore
import Capacitor

@objc public class AddDocumentResult: NSObject, Result {
    let documentReference: DocumentReference

    init(_ documentReference: DocumentReference) {
        self.documentReference = documentReference
    }

    public func toJSObject() -> AnyObject {
        var referenceResult = JSObject()
        referenceResult["id"] = self.documentReference.documentID
        referenceResult["path"] = self.documentReference.path

        var result = JSObject()
        result["reference"] = referenceResult
        return result as AnyObject
    }
}

File: packages/firestore/android/src/main/java/app/capgo/capacitor/firebase/firestore/AddDocumentResult.java
package app.capgo.capacitor.firebase.firestore.classes.results;

import app.capgo.capacitor.firebase.firestore.interfaces.Result;
import com.getcapacitor.JSObject;
import com.google.firebase.firestore.DocumentReference;

public class AddDocumentResult implements Result {

    private DocumentReference documentReference;

    public AddDocumentResult(DocumentReference documentReference) {
        this.documentReference = documentReference;
    }

    public JSObject toJSObject() {
        JSObject referenceResult = new JSObject();
        referenceResult.put("id", documentReference.getId());
        referenceResult.put("path", documentReference.getPath());

        JSObject result = new JSObject();
        result.put("reference", referenceResult);
        return result;
    }
}
